src/graph/Graph.tsx

```diff
@@ -40,6 +42,127 @@ const Graph2 = ({ data, config /* , chromaticScale */ }: Props) => {
   }
   const drawerData = tables?.find(table => table.id === drawerNodeId)
 
+  const [_highlightedColumns, _setHighlightedColumns] = useState<{
+    selectedTableId: string
+    selectedColumnId: string
+    reletedColumnsIds: Array<string>
+    reletedTablesIds: Array<string>
+  }>({
+    selectedTableId: '',
+    selectedColumnId: '',
+    reletedColumnsIds: [],
+    reletedTablesIds: [],
+  })
+
+  const setHighlightedColumns = (columnId: string, tableId: string) => {
+    if (columnId === _highlightedColumns.selectedColumnId) {
+      return _setHighlightedColumns({
+        selectedTableId: '',
+        selectedColumnId: '',
+        reletedColumnsIds: [],
+        reletedTablesIds: [],
+      })
+    }
+
+    let tableIds: Array<string> = []
+    let columnDependcies: Array<string> = []
+
+    if (highlight === 'parents') {
+      const collectColumnAndTableIds = (tId: string, cId: string): any => {
+        let tableQueue = [tId]
+        let columnsResult = [cId]
+        let tableResult: Array<string | undefined> = []
+
+        while (tableQueue.length) {
+          const currentTableId = tableQueue.shift()
+
+          tableResult.push(currentTableId)
+
+          const currentTableColumnsDependencies = tables
+            ?.find(table => table.id === currentTableId)
+            ?.columns.filter(column => columnsResult.some(cr => column.dependencies?.includes(cr)))
+            .map(column => column.id)
+            .filter(notEmpty)
+          columnsResult.push(
+            ...(currentTableColumnsDependencies ? currentTableColumnsDependencies : [])
+          )
+
+          graph.edges
+            .filter(edge => edge.target === currentTableId)
+            .forEach(edge => tableQueue.push(edge.source))
+        }
+
+        return {
+          tableIds: tableResult.filter(id => id !== tId),
+          columnIds: columnsResult.filter(id => id !== cId),
+        }
+      }
+
+      const tableAndColumnsIds = collectColumnAndTableIds(tableId, columnId)
+
+      tableIds = tableAndColumnsIds.tableIds
+      columnDependcies = tableAndColumnsIds.columnIds
+    } else if (highlight === 'children') {
+      const collectColumnAndTableIds = (tId: string, cId: string): any => {
```

Comment:
Let me suggest refactoring these duplicate parts out and moving the highlight function outside of the component